devlib/module/cgroups.py

```diff
@@ -32,157 +30,230 @@ def __new__(cls, arg):
+        for cgroup in self._cgroups:
+            if cgroup != dest:
+                self.move_tasks(cgroup, dest)
 
-class CpusetGroup(object):
+class CGroup(object):
 
-    def __init__(self, controller, name, cpus, mems):
-        self.controller = controller
+    def __init__(self, controller, name, create=True):
+        self.logger = logging.getLogger('cgroups.' + controller.kind)
         self.target = controller.target
+        self.controller = controller
         self.name = name
-        if name == 'root':
-            self.directory = controller.mount_point
-        else:
-            self.directory = self.target.path.join(controller.mount_point, name)
-            self.target.execute('mkdir -p {}'.format(self.directory), as_root=True)
-        self.cpus_file = self.target.path.join(self.directory, 'cpuset.cpus')
-        self.mems_file = self.target.path.join(self.directory, 'cpuset.mems')
-        self.tasks_file = self.target.path.join(self.directory, 'tasks')
-        self.set(cpus, mems)
-
-    def set(self, cpus, mems):
-        if isiterable(cpus):
-            cpus = list_to_ranges(cpus)
-        if isiterable(mems):
-            mems = list_to_ranges(mems)
-        self.target.write_value(self.cpus_file, cpus)
-        self.target.write_value(self.mems_file, mems)
-
-    def get(self):
-        cpus = self.target.read_value(self.cpus_file)
-        mems = self.target.read_value(self.mems_file)
-        return (cpus, mems)
 
-    def get_tasks(self):
-        task_ids = self.target.read_value(self.tasks_file).split()
-        return map(int, task_ids)
-
-    def add_tasks(self, tasks):
-        for tid in tasks:
-            self.add_task(tid)
+        # Control cgroup path
+        self.directory = controller.mount_point
+        if name != '/':
+            self.directory = self.directory + name
 
-    def add_task(self, tid):
-        self.target.write_value(self.tasks_file, tid, verify=False)
+        # Setup path for tasks file
+        self.tasks_file = self.target.path.join(self.directory, 'tasks')
+        self.procs_file = self.target.path.join(self.directory, 'cgroup.procs')
 
+        if not create:
+            return
 
-class CpusetController(CgroupController):
+        self.target.execute('[ -d {0} ] || mkdir -p {0}'\
+                .format(self.directory), as_root=True)
 
-    name = 'cpuset'
+    def exists(self):
+        try:
+            self.target.execute('[ -d {0} ]'\
+                .format(self.directory))
+            return True
+        except TargetError:
+            return False
 
-    def __init__(self, *args, **kwargs):
-        super(CpusetController, self).__init__(*args, **kwargs)
-        self.groups = {}
+    def get(self):
+        conf = {}
+
+        logging.debug('Reading %s attributes from:',
+                self.controller.kind)
+        logging.debug('  %s',
+                self.directory)
+        output = self.target.execute('grep \'\' {}/{}.*'.format(
```

Comment:
`grep` should be invoked via `self.target.busybox` as it will not present on some devices (most Android systems).